Fix ESLint warnings

We have a few warnings from ESLint that we should fix. Details below.

<details>
<summary>Logs</summary>

```
warn ESLintError:
/Users/raisedadead/DEV/freeCodeCamp/client/src/components/Header/components/NavLinks.js
  15:9   warning  Non-interactive elements should not be assigned interactive roles
jsx-a11y/no-noninteractive-element-to-interactive-role
  17:34  warning  Non-interactive elements should not be assigned interactive roles
jsx-a11y/no-noninteractive-element-to-interactive-role
  22:35  warning  Non-interactive elements should not be assigned interactive roles
jsx-a11y/no-noninteractive-element-to-interactive-role
  27:38  warning  Non-interactive elements should not be assigned interactive roles
warn ESLintError:
/Users/raisedadead/DEV/freeCodeCamp/client/src/components/search/searchBar/NoHitsSuggestion.js
  6:5  warning  Static HTML elements with event handlers require a role
jsx-a11y/no-static-element-interactions

✖ 1 problem (0 errors, 1 warning)
warn ESLintError:
/Users/raisedadead/DEV/freeCodeCamp/client/src/components/settings/DeleteModal.js
  20:7  warning  The autoFocus prop should not be used, as it can reduce usability and accessibility for
users  jsx-a11y/no-autofocus

✖ 1 problem (0 errors, 1 warning)
warn ESLintError:
/Users/raisedadead/DEV/freeCodeCamp/client/src/components/settings/ResetModal.js
  18:7  warning  The autoFocus prop should not be used, as it can reduce usability and accessibility for
users  jsx-a11y/no-autofocus

✖ 1 problem (0 errors, 1 warning)
warn ESLintError:
/Users/raisedadead/DEV/freeCodeCamp/client/src/templates/Challenges/components/Test-Suite.js
  40:13  warning  `tabIndex` should only be declared on interactive elements
jsx-a11y/no-noninteractive-tabindex

✖ 1 problem (0 errors, 1 warning)
```

</details>

--- a/client/src/components/Header/components/NavLinks.js
+++ b/client/src/components/Header/components/NavLinks.js
@@ -10,21 +10,18 @@ const propTypes = {
 function NavLinks({ displayMenu }) {
   return (
     <div className='main-nav-group'>
-      <ul
-        className={'nav-list' + (displayMenu ? ' display-flex' : '')}
-        role='menu'
-      >
-        <li className='nav-news' role='menuitem'>
+      <ul className={'nav-list' + (displayMenu ? ' display-flex' : '')}>
+        <li className='nav-news'>
           <Link external={true} sameTab={true} to='/news'>
             /news
           </Link>
         </li>
-        <li className='nav-forum' role='menuitem'>
+        <li className='nav-forum'>
           <Link external={true} sameTab={true} to='/forum'>
             /forum
           </Link>
         </li>
-        <li className='nav-projects' role='menuitem'>
+        <li className='nav-projects'>
           <Link to='/learn'>/learn</Link>
         </li>
       </ul>

--- a/client/src/components/search/searchBar/NoHitsSuggestion.js
+++ b/client/src/components/search/searchBar/NoHitsSuggestion.js
@@ -7,6 +7,7 @@ const NoHitsSuggestion = ({ title, handleMouseEnter, handleMouseLeave }) => {
       className={'no-hits-footer fcc_suggestion_item'}
       onMouseEnter={handleMouseEnter}
       onMouseLeave={handleMouseLeave}
+      role='region'
     >
       <span className='hit-name'>{title}</span>
     </div>

--- a/client/src/components/settings/DeleteModal.js
+++ b/client/src/components/settings/DeleteModal.js
@@ -17,7 +17,6 @@ function DeleteModal(props) {
   return (
     <Modal
       aria-labelledby='modal-title'
-      autoFocus={true}
       backdrop={true}
       bsSize='lg'
       className='text-center'

--- a/client/src/components/settings/ResetModal.js
+++ b/client/src/components/settings/ResetModal.js
@@ -15,7 +15,6 @@ function ResetModal(props) {
   return (
     <Modal
       aria-labelledby='modal-title'
-      autoFocus={true}
       backdrop={true}
       bsSize='lg'
       className='text-center'

--- a/client/src/templates/Challenges/components/Test-Suite.js
+++ b/client/src/templates/Challenges/components/Test-Suite.js
@@ -37,7 +37,6 @@ function TestSuite({ tests }) {
             aria-label={getAccessibleText(err, pass, text)}
             className='test-result'
             key={text.slice(-6) + index}
-            tabIndex='0'
           >
             <div className='test-status-icon'>
               {isInitial ? <Initial /> : statusIcon}

